server/src/converter.js

```diff
@@ -0,0 +1,59 @@
+const fs = require('fs');
+const path = require('path');
+const puppeteer = require('puppeteer');
+const handlebars = require('handlebars');
+
+async function createPDF(data) {
+  const templateHtml = fs.readFileSync(
+    path.join(process.cwd(), 'pdf-template.html'),
+    'utf8'
+  );
+  const template = handlebars.compile(templateHtml);
+  const html = template(data);
+
+  const pdfPath = path.join('pdf', `${data.name}.pdf`);
+
+  const options = {
+    format: 'A4',
+    headerTemplate: '<p></p>',
+    footerTemplate: '<p></p>',
+    displayHeaderFooter: false,
+    margin: {
+      top: '40px',
+      bottom: '100px',
+    },
+    printBackground: true,
+    path: pdfPath,
+  };
+
+  try {
+    const browser = await puppeteer.launch({
+      args: ['--no-sandbox'],
+      headless: true,
+    });
+
+    const page = await browser.newPage();
+
+    await page.setContent(`${html}`, {
+      waitUntil: 'networkidle0',
+    });
+
+    await page.pdf(options);
+    await browser.close();
+  } catch (err) {
+    console.log(err);
+  }
+}
+
+// pull data from db for the template
+const data = {
+  title: 'A new Resume',
+  date: '08/06/2020',
+  name: 'Volkan Van',
+  age: 39,
+  birthdate: '01/04/1981',
+  course: 'Computer Science',
+  summary: 'Summary of what I have done so far',
+};
+
+createPDF(data);
```

Comment:
Do we need to delete the pdf after we've generated it?

Reply:
I assume the file can be saved in s3 or any other storage for faster and cheaper loads instead of constructing the file every time the user wants to view it. something to explore in the future.